### PR TITLE
Cross Cluster Search: preserve remote status code

### LINKED ISSUE #27461: CCS controlling node returns a 500 when the data nodes return a 404
I have a CCS environment setup, using [the guide here](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-cross-cluster-search.html). It contains two data nodes and one controlling node. Both data nodes contain the same index, `test`, with a single document in each. One data node is running off `:9201`, the other off `:9202`. The controlling node is running off `:9220`.

This request is good:
```
curl -XPOST localhost:9201/t/_search -i
HTTP/1.1 404 Not Found
```

However, this request is unexpected:
```
curl -XPOST localhost:9220/*:t/_search -i
HTTP/1.1 500 Internal Server Error
```
(`curl -XPOST localhost:9220/cluster_one:t/_search -i` and `curl -XPOST localhost:9220/cluster_two:t/_search -i` return the same 500 too.)

I'm assuming the connecting node is detecting the 404 from the data nodes and is converting that into a 500, but I'm not sure it should do that. The connecting node should probably inspect the type of error and return the same error type instead of the 500.

### PR DESCRIPTION
**Pull request by javanna**
In case an error is returned when calling search_shards on a remote
cluster, which will lead to throwing an exception in the coordinating
 node, we should make sure that the status code returned by the
 coordinating node is the same as the one returned by the remote
 cluster. Up until now a 500 - Internal Server Error was always
 returned. This commit changes this behaviour so that for instance if an
 index is not found, which causes an 404, a 404 is also returned by the
 coordinating node to the client.

 Closes #27461

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java [hunk 1]
```diff
@@ -215,7 +215,7 @@ public void collectSearchShards(IndicesOptions indicesOptions, String preference
                                     ActionListener<Map<String, ClusterSearchShardsResponse>> listener) {
         final CountDown responsesCountDown = new CountDown(remoteIndicesByCluster.size());
         final Map<String, ClusterSearchShardsResponse> searchShardsResponses = new ConcurrentHashMap<>();
-        final AtomicReference<TransportException> transportException = new AtomicReference<>();
+        final AtomicReference<RemoteTransportException> transportException = new AtomicReference<>();
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java]: `RemoteTransportException` implements `ElasticsearchWrapperException`, then unwrapCause will unwrap the inner cause and status will return the inner status see https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/ElasticsearchException.java#L235

This behaviour is tricky. I think how ElasticsearchException#status works may cause problems when for instance it is thrown in the high-level REST client, I will have to dig deeper on that and probably open a follow-up PR, but I think this change is enough for cross-cluster search.

**Review by s1monw — APPROVED** [review]
LGTM

**Comment by javanna** [comment]
retest this please

**Comment by jpountz** [comment]
@javanna This may be backported now.

**Comment by javanna** [comment]
@jpountz done thanks a lot for the pings ;)

**Pull request merged**